src/multipole.F90

```diff
@@ -0,0 +1,199 @@
+
+    ! Allocate the multipole array components
+    call multipole % allocate()
+
+    ! Read in arrays
+    call read_dataset(group_id, "data", multipole % data)
+    call read_dataset(group_id, "pseudo_K0RS", multipole % pseudo_k0RS)
+    call read_dataset(group_id, "l_value", multipole % l_value)
+    call read_dataset(group_id, "w_start", multipole % w_start)
+    call read_dataset(group_id, "w_end", multipole % w_end)
+    call read_dataset(group_id, "broaden_poly", multipole % broaden_poly)
+
+    call read_dataset(group_id, "curvefit", multipole % curvefit)
+
+    ! Delete ACE pointwise data
+    call read_dataset(group_id, "n_grid", nuc % n_grid)
+
+    deallocate(nuc % energy)
+    deallocate(nuc % total)
+    deallocate(nuc % elastic)
+    deallocate(nuc % fission)
+    deallocate(nuc % nu_fission)
+    deallocate(nuc % absorption)
+
+    allocate(nuc % energy(nuc % n_grid))
+    allocate(nuc % total(nuc % n_grid))
+    allocate(nuc % elastic(nuc % n_grid))
+    allocate(nuc % fission(nuc % n_grid))
+    allocate(nuc % nu_fission(nuc % n_grid))
+    allocate(nuc % absorption(nuc % n_grid))
+
+    nuc % total = ZERO
+    nuc % absorption = ZERO
+    nuc % fission = ZERO
+
+    ! Read in new energy axis (converting eV to MeV)
+    call read_dataset(group_id, "energy_points", nuc % energy)
+    nuc % energy = nuc % energy / 1.0D6
+
+    ! Get count and list of MT tables
+    call read_dataset(group_id, "MT_count", NMT)
+    allocate(MT(NMT))
+
+    call read_dataset(group_id, "MT_list", MT)
+
+    call close_group(group_id)
+
+    accumulated_fission = .false.
+
+    ! Loop over each MT entry and load it into a reaction.
+    do i = 1, NMT
+      write(MT_string, '(I3.3)') MT(i)
+      MT_n = "/nuclide/reactions/MT" // MT_string
+
+      group_id = open_group(file_id, MT_n)
+
+      ! Each MT needs to be treated slightly differently.
+      select case (MT(i))
+        case(ELASTIC)
+          call read_dataset(group_id, "MT_sigma", nuc % elastic)
+          nuc % total = nuc % total + nuc % elastic
+        case(N_FISSION)
+          call read_dataset(group_id, "MT_sigma", nuc % fission)
+          nuc % total = nuc % total + nuc % fission
+          nuc % absorption = nuc % absorption + nuc % fission
+          accumulated_fission = .true.
+        case default
+          ! Search through all of our secondary reactions
+          do j = 1, nuc % n_reaction
+            if (nuc % reactions(j) % MT == MT(i)) then
+              ! Match found
+
+              ! Individual Fission components exist, so remove the combined
+              ! fission cross section.
+              if ( (MT(i) == N_F .OR. MT(i) == N_NF .OR. MT(i) == N_2NF &
+                    .OR. MT(i) == N_3NF) .AND. (accumulated_fission .eqv. .TRUE.)) then
+                nuc % total = nuc % total - nuc % fission
+                nuc % absorption = nuc % absorption - nuc % fission
+                nuc % fission = 0.0_8
+                accumulated_fission = .FALSE.
```

Comment:
lowercase `.and.`, `.or.`, `.true.`, `.false.`
`accumulated_fission .eqv. .TRUE.` => `accumulated_fission`